src/game/magic/spells/SpellsLvl01.cpp

```diff
@@ -56,25 +56,35 @@ void MagicSightSpell::Launch() {
 	m_hasDuration = m_launchDuration >= 0;
 	m_duration = m_hasDuration ? m_launchDuration : 0;
 	
-	ARX_SOUND_PlaySFX(g_snd.SPELL_VISION_START, &m_caster_pos);
+	bool emitsSound = !(m_flags & SPELLCAST_FLAG_NOSOUND);
+	
+	if(emitsSound) {
+		ARX_SOUND_PlaySFX(g_snd.SPELL_VISION_START, &m_caster_pos);
+	}
 	
 	if(m_caster == EntityHandle_Player) {
 		player.m_improve = true;
-		m_snd_loop = ARX_SOUND_PlaySFX_loop(g_snd.SPELL_VISION_LOOP, &m_caster_pos, 1.f);
+		if(emitsSound) {
+			m_snd_loop = ARX_SOUND_PlaySFX_loop(g_snd.SPELL_VISION_LOOP, &m_caster_pos, 1.f);
```

Comment:
could all or some of these ARX_SOUND calls get a conditional variant where you pass the bool in? yes of course its more efficient to put the if() around each call instead of doing a method call just to jump back immediately, but i dont think this would affect anything.

just easier than to have 500 if(emitsSound) clauses in each spell file.

alternatively, create a macro 

    #define conditional_call(condition, func) if(condition){func;}
    conditional_call(emitsSound, ARX_SOUND_PlaySFX_loop(g_snd.SPELL_VISION_LOOP, &m_caster_pos, 1.f))

Reply:
If I understand correctly, you'd pass a bool to a function just to tell it if it should do anything or not? That doesn't sound good.
It looks like the flag is set at the start of a spell and not anywhere else, so why not check the bit value once and use a member variable for that? Maybe see if you can use Spell base class to simplify what you want to do.

Reply:
Alternatively, `ARX_SOUND_PlaySFX_loop()` could be wrapped in a member function, say `playSoundLoop()` that checks for the member variable away from the main code, if reducing clutter is the goal.

Reply:
that playSoundLoop would do exactly what i proposed first with a function. the conditional macro thing would not add any extra function calls but make it a oneliner that is much easier to read. the files are long enough as they are

Reply:
I can't imagine reading that code a year later without having to inspect that macro and spending time on it. This doesn't feel like self-documenting code.